backends-common/cassandra/src/main/java/org/apache/james/backends/cassandra/init/CassandraModuleComposite.java

```diff
@@ -0,0 +1,45 @@
+package org.apache.james.backends.cassandra.init;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import com.google.common.collect.ImmutableList;
+import org.apache.james.backends.cassandra.components.CassandraIndex;
+import org.apache.james.backends.cassandra.components.CassandraModule;
+import org.apache.james.backends.cassandra.components.CassandraTable;
+import org.apache.james.backends.cassandra.components.CassandraType;
+
+public class CassandraModuleComposite implements CassandraModule {
+
+    private final ImmutableList<CassandraTable> tables;
+    private final ImmutableList<CassandraIndex> index;
+    private final ImmutableList<CassandraType> types;
+
+    public CassandraModuleComposite(CassandraModule... modules) {
+        tables = Arrays.stream(modules)
+            .flatMap(module -> module.moduleTables().stream())
+            .collect(Collectors.collectingAndThen(Collectors.toList(), ImmutableList::copyOf));
+        index = Arrays.stream(modules)
+            .flatMap(module -> module.moduleIndex().stream())
+            .collect(Collectors.collectingAndThen(Collectors.toList(), ImmutableList::copyOf));
+        types = Arrays.stream(modules)
```

Comment:
we definitively should introduce a ImmutableListCollector to use it everywhere.
See http://www.jayway.com/2013/11/12/immutable-list-collector-in-java-8/ for an example